snippets/ztp_example-sno.yaml

```diff
@@ -12,95 +12,97 @@ spec:
-        common: true
-        # ../policygentemplates/group-du-sno-ranGen.yaml will apply to all clusters with 'group-du-sno: ""'
-        group-du-sno: ""
-        # ../policygentemplates/example-sno-site.yaml will apply to all clusters with 'sites: "example-sno"'
-        # Normally this should match or contain the cluster name so it only applies to a single cluster
-        sites: "example-sno"
-      clusterNetwork:
-        - cidr: 1001:1::/48
-          hostPrefix: 64
-      machineNetwork:
-        - cidr: 1111:2222:3333:4444::/64
-      serviceNetwork:
-        - 1001:2::/112
-      additionalNTPSources:
-        - 1111:2222:3333:4444::2
-      # Initiates the cluster for workload partitioning. Setting specific reserved/isolated CPUSets is done via PolicyTemplate
-      # please see Workload Partitioning Feature for a complete guide.
-      cpuPartitioningMode: AllNodes
-      # Optionally; This can be used to override the KlusterletAddonConfig that is created for this cluster:
-      #crTemplates:
-      #  KlusterletAddonConfig: "KlusterletAddonConfigOverride.yaml"
-      nodes:
-        - hostName: "example-node1.example.com"
-          role: "master"
-          # Optionally; This can be used to configure desired BIOS setting on a host:
-          #biosConfigRef:
-          #  filePath: "example-hw.profile"
-          bmcAddress: "idrac-virtualmedia+https://[1111:2222:3333:4444::bbbb:1]/redfish/v1/Systems/System.Embedded.1"
-          bmcCredentialsName:
-            name: "example-node1-bmh-secret"
-          bootMACAddress: "AA:BB:CC:DD:EE:11"
-          # Use UEFISecureBoot to enable secure boot
-          bootMode: "UEFI"
-          rootDeviceHints:
-            wwn: "0x11111000000asd123"
-            # example of diskPartition below is used for image registry (check ImageRegistry.md for more details), but it's not limited to this use case
-          #        diskPartition:
-          #          - device: /dev/disk/by-id/wwn-0x11111000000asd123 # match rootDeviceHints
-          #            partitions:
-          #              - mount_point: /var/imageregistry
-          #                size: 102500
-          #                start: 344844
-
-          nodeNetwork:
+  - clusterName: "example-sno"
+    networkType: "OVNKubernetes"
+    # installConfigOverrides is a generic way of passing install-config
+    # parameters through the siteConfig.  The 'capabilities' field configures
+    # the composable openshift feature.  In this 'capabilities' setting, we
+    # remove all but the marketplace component from the optional set of
+    # components.
+    # Notes:
+    # - OperatorLifecycleManager is needed for 4.15 and later
+    # - NodeTuning is needed for 4.13 and later, not for 4.12 and earlier
+    installConfigOverrides:  "{\"capabilities\":{\"baselineCapabilitySet\": \"None\", \"additionalEnabledCapabilities\": [ \"OperatorLifecycleManager\", \"NodeTuning\" ] }}"
```

Comment:
It is more readable when presented in a pretty json format.

```json

installConfigOverrides: |
      {
        "capabilities": {
          "baselineCapabilitySet": "None",
          "additionalEnabledCapabilities": [
            "NodeTuning",
            "OperatorLifecycleManager"
          ]
        }
     }
```

Reply:
is it OK top put in JSON when this is a YAML snippet ?

Reply:
installConfigOverrides accepts the parameter as json input. There is actually no issue to have json parametrization in a yaml file.

Reply:
Can this change be also made in the source code Examples directory, I lifted the content from there ?  Maybe can have here as reference for easy readability if they copy and paste it will not break any further logic

Reply:
OCP doc is a reference for customers, while upstream is not, it is more intended for dev. Also we will cut our branch by friday at latest, so not sure we will have time to make that change and merge to the 4.16 branch. It is only about readability, nothing changes in the functionality.

Reply:
@kquinn1204 I've merged a PR to change the json format in our upstream repo, source code example is changed: https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/gitops-subscriptions/argocd/example/siteconfig/example-sno.yaml#L78

Reply:
a great improvement - great work